src/gc-mmtk.c

```diff
@@ -0,0 +1,1198 @@
+#include "gc-common.h"
+#include "mmtkMutator.h"
+#include "threading.h"
+
+// File exists in the binding
+#include "mmtk.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// ========================================================================= //
+// Julia specific
+// ========================================================================= //
+
+extern jl_value_t *cmpswap_names JL_GLOBALLY_ROOTED;
+extern const unsigned pool_sizes[];
+extern void _jl_free_stack(jl_ptls_t ptls, void *stkbuf, size_t bufsz);
+extern jl_mutex_t finalizers_lock;
+
+// FIXME: Does it make sense for MMTk to implement something similar
+// for now, just ignoring this.
+
+// Must be kept in sync with `base/timing.jl`
+#define FULL_SWEEP_REASON_SWEEP_ALWAYS_FULL (0)
+#define FULL_SWEEP_REASON_FORCED_FULL_SWEEP (1)
+#define FULL_SWEEP_REASON_USER_MAX_EXCEEDED (2)
+#define FULL_SWEEP_REASON_LARGE_PROMOTION_RATE (3)
+#define FULL_SWEEP_NUM_REASONS (4)
+
```

Comment:
Summary from review meeting: we'll introduce a `@static if` branch inside of `full_sweep_reasons` (from `timing.jl`), and then inside each branch we implement the version which is specific to each GC.

NOTE: `jl_full_sweep_reasons` is currently declared in `gc-stock.h`, and if it's not being used by the MMTk branch of `full_sweep_reasons`, then we probably don't need to define it for MMTk.

For now, let's just return an empty Julia dictionary in that function, and adjust it when MMTk implements these statistics.